pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -952,24 +955,12 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 		}
```

Comment:
It doesn't seem right to have cert validaton context depend on filenames.. mTLS doesn't use anything related to files. So why keep these alone?

Reply:
there's an open issue regarding this: https://github.com/istio/istio/issues/24921